opm/output/eclipse/Summary.cpp

```diff
@@ -0,0 +1,469 @@
+        case E::WLPTH: return accu + prodvol( state_well, tstep, WT::wat, dims ) +
+                                     prodvol( state_well, tstep, WT::oil, dims );
+
+        /* Production ratios */
+        case E::WWCT: return wwct( get_rate( sim_well, rt::wat, dims ),
+                                   get_rate( sim_well, rt::oil, dims ) );
+
+        case E::WWCTH: return wwcth( state_well, tstep );
+
+        case E::WGOR: return wgor( get_rate( sim_well, rt::gas, dims ),
+                                   get_rate( sim_well, rt::oil, dims ) );
+        case E::WGORH: return wgorh( state_well, tstep );
+
+        /* Pressures */
+        case E::WBHP: return convert( sim_well.bhp, dim::pressure, dims );
+        case E::WBHPH: return 0; /* not supported */
+
+        case E::WTHP: return convert( sim_well.thp, dim::pressure, dims );
+        case E::WTHPH: return 0; /* not supported */
+
+        /* Injection rates */
+        /* TODO: read from sim or compute (how?) */
+        /* TODO: Tests */
+        case E::WWIR: return -1 * (get_rate( sim_well, rt::wat, dims ));
+        case E::WOIR: return -1 * (get_rate( sim_well, rt::oil, dims ));
+        case E::WGIR: return -1 * (get_rate( sim_well, rt::gas, dims ));
+        case E::WWIT: return accu - get_vol( sim_well, rt::wat, dims );
+        case E::WOIT: return accu - get_vol( sim_well, rt::oil, dims );
+        case E::WGIT: return accu - get_vol( sim_well, rt::gas, dims );
+
+        case E::WWIRH: return injerate( state_well, tstep, WellInjector::WATER, dims );
+        case E::WOIRH: return injerate( state_well, tstep, WellInjector::GAS, dims );
+        case E::WGIRH: return injerate( state_well, tstep, WellInjector::OIL, dims );
+
+        case E::WWITH: return accu + injevol( state_well, tstep, WellInjector::WATER, dims );
+        case E::WOITH: return accu + injevol( state_well, tstep, WellInjector::GAS, dims );
+        case E::WGITH: return accu + injevol( state_well, tstep, WellInjector::OIL, dims );
+
+        case E::UNSUPPORTED:
+        default:
+            return 0;
+    }
+}
+
+}
+
+namespace out {
+
+Summary::Summary( const EclipseState& st, const Opm::Summary& sum ) :
+    Summary( st, sum, st.getTitle().c_str() )
+{}
+
+Summary::Summary( const EclipseState& st,
+                const Opm::Summary& sum,
+                const std::string& fname ) :
+    Summary( st, sum, fname.c_str() )
+{}
+
+static inline const double* get_dim_system( const EclipseState& es ) {
+    switch( es.getDeckUnitSystem().getType() ) {
+        case UnitSystem::UNIT_TYPE_METRIC: return metric;
+        case UnitSystem::UNIT_TYPE_FIELD: return field;
+        default: return metric;
+    }
+}
+
+Summary::Summary( const EclipseState& st,
+                  const Opm::Summary& sum,
+                  const char* file ) :
+    ecl_sum( 
+            ecl_sum_alloc_writer( 
+                file, false, true, ":",
```

Comment:
The two `bool` arguments - `fmt_output` and `unified` *should* be internalized in the `IOConfig` object:

```C++
const auto& ioConfig = st.getIOConfigConst();
bool fmt_out = ioConfig->getFMTOUT();
bool unified = ioConfig->getUNIFOUT();
...
```